alfresco-search/src/test/java/org/alfresco/solr/tracker/DistributedCascadeTrackerTest.java

```diff
@@ -0,0 +1,207 @@
+
+        // Check if the path is updated for both the nodes
+        assertShardCount(0, params("qt", "/afts", "q", "PATH:" + cascadingFirstChild), 1);
+        assertShardCount(1, params("qt", "/afts", "q", "PATH:" + cascadingSecondChild), 1);
+
+        // The old paths are no longer in the index
+        assertShardCount(0, params("qt", "/afts", "q", "PATH:" + pathChild0), 0);
+        assertShardCount(1, params("qt", "/afts", "q", "PATH:" + pathChild1), 0);
+
+    }
+
+
+    private void indexNodes(Acl acl) throws Exception{
+
+        Transaction bigTxn = getTransaction(0, 3);
+
+        /*
+         * Create parent folder in the first shard
+         */
+        parentFolder = getNode(0, bigTxn, acl, Node.SolrApiNodeStatus.UPDATED);
+        parentFolderMetadata = getNodeMetaData(parentFolder, bigTxn, acl, "elia", null, false);
+        parentFolderMetadata.setPaths(of(new Pair<>(pathParent, null)));
+
+        /*
+         * Create first node.
+         * This will be stored in the first shard (range [0-100])
+         */
+        childShard0 = getNode(99, bigTxn, acl, Node.SolrApiNodeStatus.UPDATED);
+        childShardMetadata0 = getNodeMetaData(childShard0, bigTxn, acl, "elia", ancestors(parentFolderMetadata.getNodeRef()), false);
+        childShardMetadata0.setPaths(of(new Pair<>(pathChild0, null)));
+
+        /*
+         * Create second node.
+         * This will be stored in the second shard (range [101-200])
+         */
+        childShard1 = getNode(101, bigTxn, acl, Node.SolrApiNodeStatus.UPDATED);
+        childShardMetadata1 = getNodeMetaData(childShard1, bigTxn, acl, "elia", ancestors(parentFolderMetadata.getNodeRef()), false);
+        childShardMetadata1.setPaths(of(new Pair<>(pathChild1, null)));
+
+
+        indexTransaction(bigTxn,
+                of(parentFolder, childShard0, childShard1),
+                of(parentFolderMetadata, childShardMetadata0, childShardMetadata1));
+
+
+        /*
+         * Get sure the nodes are indexed correctly in the shards
+         */
+        waitForDocCount(new TermQuery(new Term("content@s___t@{http://www.alfresco.org/model/content/1.0}content", "world")), 3, timeout);
+        assertShardCount(0, new TermQuery(new Term("content@s___t@{http://www.alfresco.org/model/content/1.0}content", "world")), 2);
+        assertShardCount(1, new TermQuery(new Term("content@s___t@{http://www.alfresco.org/model/content/1.0}content", "world")), 1);
+    }
+
+
+    private void indexParentFolderWithCascade(){
+
+        Transaction bigTxn = getTransaction(0, 1);
+
+        parentFolder.setTxnId(bigTxn.getId());
+        parentFolderMetadata.setTxnId(bigTxn.getId());
+        parentFolderMetadata.getProperties().put(ContentModel.PROP_CASCADE_TX, new StringPropertyValue(Long.toString(bigTxn.getId())));
+
+        indexTransaction(bigTxn, of(parentFolder), of(parentFolderMetadata));
+    }
+
+
+    protected static Properties getShardMethod()
```

Comment:
Why this method is static?

Reply:
It was static because the initData method was static as well. I modified initData (the method where the shards are created) in order to be non static because the shards should be clean for each test (even if currently there is only one test).